pkg/api/pin.go

```diff
@@ -49,28 +52,60 @@ func (s *Service) pinRootHash(w http.ResponseWriter, r *http.Request) {
 	getter := s.storer.Download(true)
 	traverser := traversal.New(getter)
 
+	sem := semaphore.NewWeighted(100)
+	var errTraverse error
+	var mtxErr sync.Mutex
+	var wg sync.WaitGroup
```

Comment:
You can use "golang.org/x/sync/errgroup" instead and get rid of the mutex and error.

Reply:
i thought about using that but I still need access to the internal error inside to goroutine to stop the traversal.